MainForm.cs

```diff
@@ -0,0 +1,173 @@
+﻿using EnumColors;
+using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.Windows.Forms;
+
+namespace Task1_GUI
+{
+    public partial class MainForm : Form
+    {
+        public MainForm()
+        {
+            InitializeComponent();
+        }
+
+        bool CreateCardCombination(int combLen, List<Card> cards, TextBox textBox)
+        {
+            bool isCombinationCorrect = true;
+            for (int i = 0; i < combLen; i++)
+            {
+                Card card = new Card(textBox.Lines[i + 1]);
+                isCombinationCorrect = Card.IsCardCorrect;
+                if (!isCombinationCorrect)
+                {
+                    textBox.Text = Service.ErrorMessage;
+                    textBox.ForeColor = Color.Red;
+                    break;
+                }
+                else
+                {
+                    cards.Add(card);
+                }
+            }
+            return isCombinationCorrect;
+        }
+        
+        public void MainFunc(int combLen1, int combLen2)
+        {
+            Label[] labelsOnCards1 = { c11, c12, c13, c14, c15, c16, c17 };
+            Label[] labelsOnCards2 = { c21, c22, c23, c24, c25, c26, c27 };
+            Panel[] cardPanels1 = { card11, card12, card13, card14, card15, card16, card17 };
+            Panel[] cardPanels2 = { card21, card22, card23, card24, card25, card26, card27 };
+            List<Card> cards1 = new List<Card>();
+            List<Card> cards2 = new List<Card>();
+            Card higherCard1, higherCard2;
+            bool isCombination_1_Correct, isCombination_2_Correct;
+            bool existIdenticalCards;
+
+            isCombination_1_Correct = CreateCardCombination(combLen1, cards1, textBox1);
+            isCombination_2_Correct = CreateCardCombination(combLen2, cards2, textBox2);
+
+            existIdenticalCards = Service.AreAnyIdenticalCards(cards1, cards2);
+
+            if (existIdenticalCards)
+            {
+                textBox1.Text = Service.ErrorMessage;
+                textBox1.ForeColor = Color.Red;
+                textBox2.Text = Service.ErrorMessage;
+                textBox2.ForeColor = Color.Red;
+            }
+            if (isCombination_1_Correct && isCombination_2_Correct && !existIdenticalCards)
+            {
+                cards1.Sort();
+                cards2.Sort();
+
+                higherCard1 = cards1[cards1.Count - 1];
+                higherCard2 = cards1[cards2.Count - 1];
+
+                Service.FindWinCombination(higherCard1, higherCard2, cards1, cards2);
+
+                label3.Text = "Выиграла комбинация № " + Service.WinCombination.NumOfWinner;
```

Comment:
What if somebody will forget to call the `FindWinCombination` method before? We don't want this relations. You need to return the `Card` object from the `FindWinCombination` method